src/BootAppManager.ts

```diff
@@ -1,26 +1,105 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT license.
 
-import { BootApp } from "./BootApp";
+import { BootApp, STATE_INACTIVE } from "./BootApp";
+
+import * as vscode from 'vscode';
+import * as uuid from 'uuid';
+import * as path from 'path';
+
+interface JavaProjectData {
+    path : string
+    name : string
+    classpath : ClassPathData
+}
+
+interface ClassPathData {
+    entries : CPE[];
+}
+
+interface CPE {
+    kind: string;
+    path: string; // TODO: Change to File, Path or URL?
+	outputFolder : string;
+	sourceContainerUrl : string;
+	javadocContainerUrl : string;
+	isSystem : boolean;
+}
+
+function isBootAppClasspath(cp : ClassPathData) : boolean {
+    if (cp.entries) {
+        let entries = cp.entries;
+        for (let i = 0; i < entries.length; i++) {
+            const cpe = entries[i];
+            let filename = path.basename(cpe.path);
+
+            if (filename.endsWith('.jar') && filename.startsWith('spring-boot')) {
+                return true;
+            }
+        }
+    } 
+    return false;
+}
 
 export class BootAppManager {
-    private _appList: BootApp[];
+
+    private _boot_projects : Map<String, JavaProjectData> = new Map();
 
     constructor() {
-        this._appList = [];
         this.initAppListSync();
-
     }
 
     public getAppList(): BootApp[] {
-        return this._appList;
+        let apps : BootApp[] = [];
+        this._boot_projects.forEach(p => {
+            apps.push(new BootApp(p.name, STATE_INACTIVE));
+        });
+        return apps;
     }
 
     private initAppListSync(): void {
-        // TODO: this is just a mock.
-        this._appList.push(
-            new BootApp("app 1", "running"),
-            new BootApp("app 2", "inactive")
-        );
+        //TODO: The code below will fail if jdt language server has not yet been started
```

Comment:
I don't think there's an alternative for having Java editing support. Maybe we could somehow avoid a direct dependency (not sure how we get classpath then though) but as a user, I think you probably will have to install it anyway. Working with spring boot project without Java editing support seems pretty close to useless.

Reply:
OK, so let's settle down to assume the users have language server installed now.

Reply:
The TODO wasn't actually just about whether the java server is installed, but whether it already started. This is something we have to put some thought into still as it could be a problem. What I mean is, the language server may be installed... but it isn't started unless the user opens a Java file. I wasn't really sure how / if we can deal with this. Maybe you have some ideas (for example, is there a way we can force the `redhat.java` and pivotal.spring-boot` extensions to be started before the boot dash extension? (Or is that perhaps already implied by adding them as `extensionDependencies` to package.json file?

Reply:
I briefly tested this and it seems that adding the `extensionDependencies` to `package.json` ensures the depended-on extensions are being started. So perhaps I'm worried about nothing and this TODO could be removed.

Can someone else confirm this?